ts/kpt-functions/src/metadata.ts

```diff
@@ -16,9 +16,14 @@
 
 import { KubernetesObject } from './types';
 
-export const ANNOTATION_PREFIX = 'config.kubernetes.io';
+export const ANNOTATION_PREFIX = 'internal.config.kubernetes.io';
 export const SOURCE_PATH_ANNOTATION = `${ANNOTATION_PREFIX}/path`;
 export const SOURCE_INDEX_ANNOTATION = `${ANNOTATION_PREFIX}/index`;
+export const ID_ANNOTATION = `${ANNOTATION_PREFIX}/id`;
+export const LEGACY_ANNOTATION_PREFIX = 'config.kubernetes.io';
+export const LEGACY_SOURCE_PATH_ANNOTATION = `${LEGACY_ANNOTATION_PREFIX}/path`;
+export const LEGACY_SOURCE_INDEX_ANNOTATION = `${LEGACY_ANNOTATION_PREFIX}/index`;
+export const LEGACY_ID_ANNOTATION = `${LEGACY_ANNOTATION_PREFIX}/index`;
```

Comment:
The legacy id annotation is actually `config.k8s.io/id`, while the others have the prefix `config.kubernetes.io`.